crates/ruff_linter/src/rules/flake8_comprehensions/snapshots/ruff_linter__rules__flake8_comprehensions__tests__C416_C416.py.snap

```diff
@@ -84,47 +85,6 @@ C416.py:9:1: C416 [*] Unnecessary dict comprehension (rewrite using `dict()`)
 11 11 | {k: (a, b) for k, (a, b) in d.items()}
 12 12 | 
 
-C416.py:10:1: C416 [*] Unnecessary list comprehension (rewrite using `list()`)
-   |
- 8 | {k: v for k, v in y}
- 9 | {k: v for k, v in d.items()}
-10 | [(k, v) for k, v in d.items()]
-   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ C416
-11 | {k: (a, b) for k, (a, b) in d.items()}
-   |
-   = help: Rewrite using `list()`
```

Comment:
Once type inference is supported, C416 flag this line.

Reply:
We do have limited support for telling when something is a dictionary - at least if it is defined in the same file where the lint violation occurs. See: https://github.com/astral-sh/ruff/blob/881375a8d9b30291316af70b0e1e9327b7bb9ab7/crates/ruff_python_semantic/src/analyze/typing.rs#L836 .

You can see examples of how this is used in other rules, e.g. https://github.com/astral-sh/ruff/blob/881375a8d9b30291316af70b0e1e9327b7bb9ab7/crates/ruff_linter/src/rules/flake8_simplify/rules/key_in_dict.rs#L123-L135